src/main/java/com/github/lookout/metrics/agent/generators/YammerMetricsGenerator.java

```diff
@@ -0,0 +1,87 @@
+package com.github.lookout.metrics.agent.generators;
+
+import com.timgroup.statsd.StatsDClient;
+import com.yammer.metrics.Metrics;
+import com.yammer.metrics.core.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Created by rkuris on 5/28/15.
+ */
+public class YammerMetricsGenerator implements MetricProcessor<Long>, MetricGenerator {
+
+    private static final Logger LOG = LoggerFactory.getLogger(YammerMetricsGenerator.class);
+    private MetricsRegistry registry = Metrics.defaultRegistry();
+    private StatsDClient client;
+    private static final MetricPredicate predicate = MetricPredicate.ALL;
+    protected final Clock clock = Clock.defaultClock();
+
+    @Override
+    public void generate(StatsDClient client) {
+        // get all the registered metrics via dropwizard
+        this.client = client;
+        final long epoch = TimeUnit.MILLISECONDS.toSeconds(clock.time());
+        final Set<Map.Entry<String, SortedMap<MetricName, Metric>>> entries = registry.groupedMetrics(predicate).entrySet();
+
+        for (final Map.Entry<String, SortedMap<MetricName, Metric>> entry : entries) {
+            for (final Map.Entry<MetricName, Metric> subEntry : entry.getValue().entrySet()) {
+
+                final Metric metric = subEntry.getValue();
+
+                if (metric != null) {
+                    try {
+                        metric.processWith(this, subEntry.getKey(), epoch);
+                    } catch (final Exception exception) {
+                        LOG.error("Error processing key {}", subEntry.getKey(), exception);
+                    }
+                }
+            }
+        }
+
+    }
+    @Override
+    public void processMeter(MetricName name, Metered meter, Long epoch) throws Exception {
+        LOG.debug("Meter {} {} skipped", name.getName(), meter.count());
+    }
+
+    @Override
+    public void processCounter(MetricName name, Counter counter, Long epoch) throws Exception {
+        client.gauge(name.getName(), counter.count());
```

Comment:
You are right. Grand scheme of things, if we are looking at particular metrics and they are unique enough, then it should be ok. 

If we have an MBean+Atttribute as follows:
`org.apache.cassandra.metrics:type=ColumnFamily,keyspace=system,scope=peer_events,name=EstimatedColumnCount.Value`
We are reproting it as:
`EstimatedColumnCount.Value`
Instead, what if we report it as:
`org.apache.cassandra.metrics.ColumnFamily.system.peer_events.EstimatedColumnCount.Value`

In any case, this wont be available until we use jmxtrans-agent; but something to keep an eye on.